include/aws/common/hash_table.h

```diff
@@ -318,6 +318,18 @@ int aws_hash_table_foreach(
     int (*callback)(void *context, struct aws_hash_element *p_element),
     void *context);
 
+/**
+ * Compares two hash tables for equality. Both hash tables must have equivalent
+ * key comparators; values will be compared using the comparator passed into this
+ * function. The key hash function does not need to be equivalent between the
+ * two hash tables.
+ */
+AWS_COMMON_API
+bool aws_hash_table_eq(
+    bool (*value_eq)(const void *a, const void *b),
```

Comment:
Just curious, why is this the first parameter?

Reply:
I'm happy with it being either first or last. Which do you think is more consistent?

Reply:
My vote is last, looking at things like C++'s [`std::equal`](https://en.cppreference.com/w/cpp/algorithm/equal), the predicate is always last.